### PR TITLE
Support `ByteStream`s in `bytes starts-with` and `bytes ends-with`

### PR DESCRIPTION
**Pull request by IanManske**
# Description
Restores `bytes starts-with` so that it is able to work with byte streams once again. For parity/consistency, this PR also adds byte stream support to `bytes ends-with`.

# User-Facing Changes
- `bytes ends-with` now supports byte streams.

# Tests + Formatting
Re-enabled tests for `bytes starts-with` and added tests for `bytes ends-with`.

### CODE REVIEW

File: crates/nu-command/src/bytes/ends_with.rs [hunk 1]
```diff
@@ -52,14 +56,49 @@ impl Command for BytesEndsWith {
...
+            let mut end = VecDeque::<u8>::with_capacity(cap);
+            loop {
+                let buf = match reader.fill_buf() {
+                    Ok(&[]) => break,
+                    Ok(buf) => buf,
+                    Err(e) if e.kind() == io::ErrorKind::Interrupted => continue,
+                    Err(e) => return Err(e.into_spanned(stream_span).into()),
+                };
+                let len = buf.len();
+                if len >= cap {
```
Comment (WindSoilder) [inline crates/nu-command/src/bytes/ends_with.rs:82]: Maybe we don't need this `if` part?  To me it seems that `else` part handles for both cases

**Review by WindSoilder — CHANGES_REQUESTED** [review]
Hi, Thank you for your pr!  It makes me understand more about the usage of ByteStream.

I leave some thought.

File: crates/nu-command/src/bytes/starts_with.rs [hunk 2]
```diff
@@ -53,20 +54,46 @@ impl Command for BytesStartsWith {
...
+                };
+                let len = buf.len();
+                if len >= pattern.len() {
+                    break buf.starts_with(pattern);
+                }
+                let (pat, remaining) = pattern.split_at(len);
+                if buf != pat {
+                    break false;
+                }
+                reader.consume(len);
```
[the remark this answers is not in the document]
Comment (IanManske) [reply crates/nu-command/src/bytes/starts_with.rs:86]: Thanks, this is much simpler!

File: crates/nu-command/src/bytes/ends_with.rs (hunk 1 above)
Comment (IanManske) [reply crates/nu-command/src/bytes/ends_with.rs:82]: The issue is that `drain` will panic if the range is out of the bounds. I will add a comment.

**Review by WindSoilder — APPROVED** [review]
Thanks!  It looks good to me!

**Pull request merged**